src/test/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,104 @@
+package nearsoft.academy.bigdata.recommendation;
+
+import org.apache.commons.collections.functors.WhileClosure;
+import org.apache.commons.math.stat.descriptive.summary.Product;
+
+import java.io.BufferedReader;
+import java.util.*;
+import java.io.FileReader;
+
+public class MovieRecommender {
+
+
+    ArrayList<Review> listaDeReviews = new ArrayList<>();
+    int contadorDeReviews = 0;
+    public MovieRecommender(String filepath ) {
+
+       try{
+            FileReader fr = new FileReader(filepath);
+            BufferedReader br = new BufferedReader(fr);
+
+            String nuevaLinea;
+
+            Review objeto = new Review();
+            while ((nuevaLinea = br.readLine()) != null) {
+
+                String words[] = nuevaLinea.split(" ");
+
+                if (words[0].equals("product/productId:")) {
+                    objeto.setProductId(words[1]);
+                }
+
+                if (words[0].equals("review/userId:")) {
+                    objeto.setUserId(words[1]);
+                }
+
+                if (nuevaLinea.isEmpty()) {
```

Comment:
This if should be the first validation, and all the code of the while should be refactored with "else if" and "returns" to avoid multiple validations